federation/pkg/federation-controller/secret/secret_controller.go

```diff
@@ -368,6 +368,11 @@ func (s *FederationSyncController) reconcile(namespacedName types.NamespacedName
 	}
 	obj := copiedObj.(pkgruntime.Object)
 	meta := s.adapter.ObjectMeta(obj)
+	if util.IsFederationOnlyObject(obj) {
+		// Should not reconcile federation-only objects
+		glog.V(4).Infof("Skipping federation only Secret: %s", key)
```

Comment:
This is not actually the secret controller anymore.  It's a generic sync controller and the type being federated is indicated via the ``kind`` variable:

```golang
glog.V(4).Infof("Skipping federation-only %q: %s", kind, key)
```

Also, grammar police: ``federation-only`` matches your intent.  ``federation only`` does not.